designer/server/src/routes/file/file.js

```diff
@@ -0,0 +1,97 @@
+import Boom from '@hapi/boom'
+import { StatusCodes } from 'http-status-codes'
+import Joi from 'joi'
+
+import { redirectWithErrors } from '../forms/create.js'
+
+import { sessionNames } from '~/src/common/constants/session-names.js'
+import { checkFileStatus, createFileLink } from '~/src/lib/file.js'
+import { errorViewModel } from '~/src/models/errors.js'
+import * as file from '~/src/models/file/file.js'
+
+export const emailSchema = Joi.string().trim().required().messages({
+  'string.empty': 'Enter an email address'
+})
+
+export default [
+  /**
+   * @satisfies {ServerRoute< { Params:  { fileId: string } }>}
+   */
+  ({
+    method: 'GET',
+    path: '/file-download/{fileId}',
+    async handler(request, h) {
+      const { params, yar } = request
+      const { fileId } = params
+
+      const statusCode = await checkFileStatus(fileId)
+
+      switch (statusCode) {
+        case StatusCodes.OK: {
+          const validation = yar.flash(sessionNames.validationFailure)[0]
+          return h.view('file/download-page', file.fileViewModel(validation))
```

Comment:
there is below tslint error on this line. Need help please 

`file.js(32,66): error TS2345: Argument of type 'ValidationFailure<FormMetadataInput>' is not assignable to parameter of type 'ValidationFailure<{ email: string; }>'.`

Reply:
Have a look at the types for `yar` in `designer/server/src/types.ts` line 93. I think that's where the problem is. It doesn't know that `const validation = yar.flash(sessionNames.validationFailure)[0]` can return what you want it to in the next line `file.fileViewModel(validation)`.

Reply:
I tried few things in https://github.com/DEFRA/forms-designer/pull/409/commits/20f00ddf43b2777b6290e5f7f32718591f9f708e but couldn't get it to work. I just defined as object which is not great, i will leave it to that unless someone has better suggestion.